src/client/components/app/Transactions/RepayPositionTx.tsx

```diff
@@ -558,66 +676,144 @@ export const RepayPositionTx: FC<RepayPositionProps> = (props) => {
+              inputText={claimTokenHeaderText}
+              // TODO: create unit test for this
+              amount={sellToken === buyToken ? claimTargetBalance : targetAmount}
+              onAmountChange={(amnt) => setTargetAmount(amnt)}
               // token to claim from spigot
               selectedToken={selectedSellToken}
               // 0x testing data
               // selectedToken={tokensMap[ETH]}
               onSelectedTokenChange={onSelectedSellTokenChange}
-              tokenOptions={sourceAssetOptions} // TODO get options from unusedToken data in subgraph
-              readOnly={true}
+              tokenOptions={claimableTokenOptions}
+              readOnly={sellToken === buyToken ? true : false}
             />
-            {/* redner bought tokens based on trade data from 0x API response */}
-            {!!tokensToBuy ? (
-              <TxTokenInput
-                headerText={t('components.transaction.repay.claim-and-repay.credit-token')}
-                inputText={t('components.transaction.repay.claim-and-repay.buy-amount')}
-                amount={normalizeAmount(tokensToBuy, selectedPosition.token.decimals)}
-                selectedToken={selectedPosition.token}
-                // 0x testing data
-                // selectedToken={tokensMap[DAI]}
-                readOnly={true}
-              />
-            ) : (
-              <TradeError> {t('components.transaction.repay.claim-and-repay.insufficient-liquidity')} </TradeError>
-            )}
+            {/* rendner tokens to purchase based on trade data from 0x API response */}
+            {sellToken !== buyToken ? (
+              <>
+                <TxTokenInput
+                  headerText={t('components.transaction.repay.claim-and-repay.credit-token')}
+                  inputText={t('components.transaction.repay.claim-and-repay.buy-amount')}
+                  // TODO: create unit test for this
+                  amount={tokensToBuy} // TODO: uncomment this after testing 0x trade data.
+                  // amount={targetAmount} // TODO: delete this after confirming 0x trade data is working.
+                  selectedToken={selectedPosition.token}
+                  // 0x testing data
+                  // selectedToken={tokensMap[DAI]}
+                  readOnly={true}
+                />
+                {/* TODO: Determine how to display an insufficient liquidity message when testing on Ethereum mainnet. */}
+                {/* {zeroExWarning ? (
+                  <TradeError> {t('components.transaction.repay.claim-and-repay.insufficient-liquidity')} </TradeError>
+                ) : null} */}
+              </>
+            ) : null}
           </>
         );
 
-      case 'deposit-and-close':
       case 'close':
+        return (
+          <TxTokenInput
+            headerText={t('components.transaction.repay.select-amount')}
+            inputText={tokenHeaderText}
+            amount={getInterestAccrued()}
```

Comment:
i feel like `getInterestAccrued()` should be called on render in a useEffect and it should update position data in state.

Reply:
and then use position.interestAccrued here like normal

Reply:
this is still a work in progress. need to deploy new contracts with view funcion. For now, we are approving max int so borrower can pay remaining interest even though we dont know how much

Reply:
TODO